pkg/cmd/auth/gitcredential/helper.go

```diff
@@ -100,12 +100,18 @@ func helperRun(opts *CredentialOptions) error {
 		return err
 	}
 
+	lookupHost := wants["host"]
 	var gotUser string
-	gotToken, source, _ := cfg.GetWithSource(wants["host"], "oauth_token")
+	gotToken, source, _ := cfg.GetWithSource(lookupHost, "oauth_token")
+	if gotToken == "" && strings.HasPrefix(lookupHost, "gist.") {
+		lookupHost = strings.TrimPrefix(lookupHost, "gist.")
+		gotToken, source, _ = cfg.GetWithSource(lookupHost, "oauth_token")
+	}
```

Comment:
Will this work for enterprise? From the `GistHost` function above it seems like the enterprise hosts won't follow the `gist.` prefix pattern.

Reply:
Also seems like we could safely always trim the `gist.` prefix before doing the lookup.

Reply:
`gist.` should also be the prefix on Enterprise hosts in general, but in cases that a subdomain is not available, `mygithub.org/gist/*` also works (and otherwise redirects to the `gist.` subdomain)